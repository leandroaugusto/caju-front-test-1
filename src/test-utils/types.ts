export interface ITestUtils {
  children?: React.ReactNode
}
