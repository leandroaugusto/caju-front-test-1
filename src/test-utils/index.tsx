import { ReactElement } from "react";
import { ThemeProvider } from "styled-components";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";

import { ITestUtils } from "./types";

import theme from "~/theme";
import type * as React from "react";

export const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      gcTime: Infinity,
    },
  },
});

export const QueryProvider = ({ children }: ITestUtils) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

const AllTheProviders: React.FC = ({ children }: ITestUtils) => {
  return (
    <QueryProvider>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </QueryProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  });

export * from "@testing-library/react";

export { customRender };
