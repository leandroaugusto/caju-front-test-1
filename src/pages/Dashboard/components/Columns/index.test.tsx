import { vi } from "vitest";
import { customRender, screen } from "~/test-utils";

import { Columns } from ".";
import {
  TRegistrationsData,
  ERegistrationsStatus,
} from "~/types/registrations.types";

const registrationsMock: TRegistrationsData[] = [
  {
    admissionDate: "22/10/2023",
    email: "luiz@caju.com.br",
    employeeName: "Luiz Filho",
    status: ERegistrationsStatus.APPROVED,
    cpf: "56642105087",
    id: "3",
  },
  {
    admissionDate: "22/10/2022",
    email: "luiz@caju.com.br",
    employeeName: "Luiz Filho Jr",
    status: ERegistrationsStatus.REVIEW,
    cpf: "56642105088",
    id: "2",
  },
];

vi.mock("../RegistrationCard", () => ({
  RegistrationCard: ({ data }: { data: TRegistrationsData }) => (
    <div data-testid="mock-card-container">{data.status}</div>
  ),
}));

describe("Columns", () => {
  it("Should render Columns component", () => {
    const { container } = customRender(
      <Columns registrations={registrationsMock} />
    );

    const columnsTitle = screen.getAllByTestId("column-title");
    const columnsContent = screen.getAllByTestId("column-content");
    const cardContainer = screen.getAllByTestId("mock-card-container");

    expect(columnsTitle).toHaveLength(3);
    expect(columnsContent).toHaveLength(3);
    expect(cardContainer).toHaveLength(2);

    expect(columnsContent[0]).toContainElement(cardContainer[0]);
    expect(columnsContent[1]).toContainElement(cardContainer[1]);
    expect(columnsContent[2]).toBeEmptyDOMElement();

    expect(container).toMatchSnapshot();
  });
});
