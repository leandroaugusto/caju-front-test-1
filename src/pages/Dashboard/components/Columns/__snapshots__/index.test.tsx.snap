// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Columns Should render Columns component 1`] = `
<div>
  <div
    class="sc-braxZu iMmUiR"
    data-testid="columns-container"
  >
    <div
      class="sc-gJhJTp HbPSK"
    >
      <h3
        class="sc-elDIKY eGvgBu"
        data-testid="column-title"
      >
        Pronto para revisar
      </h3>
      <div
        class="sc-fQpRED klzfkN"
        data-testid="column-content"
      >
        <div
          data-testid="mock-card-container"
        >
          REVIEW
        </div>
      </div>
    </div>
    <div
      class="sc-gJhJTp fUdUew"
    >
      <h3
        class="sc-elDIKY duOeG"
        data-testid="column-title"
      >
        Aprovado
      </h3>
      <div
        class="sc-fQpRED klzfkN"
        data-testid="column-content"
      >
        <div
          data-testid="mock-card-container"
        >
          APPROVED
        </div>
      </div>
    </div>
    <div
      class="sc-gJhJTp kmWUQj"
    >
      <h3
        class="sc-elDIKY cLMMBy"
        data-testid="column-title"
      >
        Reprovado
      </h3>
      <div
        class="sc-fQpRED klzfkN"
        data-testid="column-content"
      />
    </div>
  </div>
</div>
`;
